### ISSUE TITLE
SyncBatchNorm should avoid sync when model.eval() is True.

### ISSUE THREAD

**Issue by pritamdamania87** [pytorch/pytorch#48988]
## 🚀 Feature
SyncBatchNorm should avoid sync when model.eval() is True.

## Motivation

Similar to how BatchNorm layers operate differently in eval mode, SyncBatchNorm probably shouldn't synchronize stats across multiple ranks and just behave as a regular BatchNorm layer in this model. This was motivated by the following forum post: https://discuss.pytorch.org/t/validation-hangs-up-when-using-ddp-and-syncbatchnorm/104831/7.

## Pitch

SyncBatchNorm avoids sync during the eval phase and operates like a regular BatchNorm layer.

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @agolynski @SciPioneer @H-Huang @mrzzd

**Comment by ParthaEth** [comment]
Any updates on this?

**Comment by rohan-varma** [comment]
@pritamdamania87 If we disable all sync for model.eval() in syncBN, is it technically BC breaking as the user may get different results (stats won't be synced across processes anymore)? If so, do we need to provide a flag that is off by default to do this?

**Comment by pritamdamania87** [comment]
@rohan-varma I think it is a bug that stats are synchronized even after calling `.eval()` so we shouldn't worry about breaking something that was already buggy.
